inc/vcf/fixer.hpp

```diff
@@ -108,9 +109,48 @@ namespace ebi
             util::writeline(output, *line);
             ignored_errors++;
         }
+        /**
+         * fix: remove the info field that caused the error
+         */
         virtual void visit(InfoBodyError &error) override
         {
-            ignored_errors++;
+            // TODO better log system, if any
+            std::cerr << "DEBUG: line " << error.get_line() << ": fixing invalid INFO field " << error.get_field()
+                      << std::endl;
+            int fixed = 0;
+
+            std::vector<std::string> columns;
+            std::string string_line{line->begin(), line->end()};
+            util::string_split(string_line, "\t", columns);
+            std::vector<std::string> infos;
+            util::string_split(columns[7], ";", infos);
+
+            for (size_t i = 0; i < 7; ++i) {
+                output << columns[i] << "\t";
+            }
+            std::vector<std::string> key_value;
+            for (size_t j = 0; j < infos.size(); ++j) {
+                util::string_split(infos[j], "=", key_value);
+                if (key_value[0] != error.get_field()) {
+                    if ((fixed - j) != 0) { // if this is not the first field we don't remove, then write the separator
+                        output << ";";
+                    }
+                    output << infos[j];
+                } else {
+                    fixed++;
+                }
+            }
+
+            if (fixed == infos.size()) {
+                output << ".";
```

Comment:
just like the column_number, I'd rather extract it as a constant string, because an empty field may be expressed differently across fields. however, in this case I don't mind adding a `write_empty_field` function, as it's more likely that `.` does apply as empty field in every column. So, function or const string?